smf/ntp/manifest/manifest.xml

```diff
@@ -1,105 +1,93 @@
-                    type="service">
-            <service_fmri value="svc:/network/routing-setup" />
-        </dependency>
-
-	<!--
-        The service is started as root, but with only the privileges it
-	    requires. The chrony daemon forks a privileged helper and then the
-	    main daemon drops everything it does not need, leaving it with
-	    completely minimal privileges (it even divests the privilege to
-        fork/exec).
-
-        The service also always starts the binary with ASLR enabled,
-	    regardless of whether it was linked with -zaslr
-	-->
-        <exec_method type="method"
-                     name="start"
-		     exec="/var/svc/method/svc-site-ntp %m"
-                     timeout_seconds="60">
-            <method_context security_flags="aslr">
-                <method_credential user="root"
-                                   group="root"
-                                   privileges="basic,!file_link_any,!proc_info,!proc_session,file_chown_self,file_dac_search,file_dac_write,net_privaddr,proc_lock_memory,proc_priocntl,proc_setid,sys_time" />
-            </method_context>
-        </exec_method>
-
-        <exec_method type="method"
-                     name="refresh"
-		     exec="/var/svc/method/svc-site-ntp %m %{restarter/contract}"
-                     timeout_seconds="60">
-        </exec_method>
-
-        <exec_method type="method"
-                     name="stop"
-		     exec="/var/svc/method/svc-site-ntp %m %{restarter/contract}"
-                     timeout_seconds="60" />
-
-        <property_group name="config"
-                        type="application">
-            <propval name="file"
-                     type="astring"
-                     value="/etc/inet/chrony.conf" />
-	    <propval name="boundary"
-                     type="boolean"
-		    value="false" />
-            <propval name="server"
-                     type="astring"
-                     value="" />
-            <propval name="allow"
-                     type="astring"
-                     value="" />
-        </property_group>
-
-        <stability value="Unstable" />
-
-        <template>
-            <common_name>
-                <loctext xml:lang="C">Network Time Protocol (NTP)</loctext>
-            </common_name>
-        </template>
-
-    </service>
+
+<service_bundle type="manifest" name="ntp:default">
+
+  <service name="oxide/ntp" type="service" version="1">
+  <create_default_instance enabled="true" />
+
+  <dependency name="network" grouping="require_any" restart_on="error"
+    type="service">
+  <service_fmri value="svc:/network/service" />
+  </dependency>
+
+  <dependency name="filesystem" grouping="require_all" restart_on="error"
+    type="service">
+  <service_fmri value="svc:/system/filesystem/minimal" />
+  </dependency>
+
+  <dependency name="name-services" grouping="optional_all" restart_on="none"
+    type="service">
+  <service_fmri value="svc:/milestone/name-services" />
```

Comment:
Note that we have a dependency on `milestone/name-services`, which means we wait until the `network/dns/client` service is online. `network/dns/client` depends on both `dns/install` and the presence of `/etc/resolv.conf`, so everything works nicely and we'll block waiting for those.

I'd change `optional_all` here to `require_all` to make this more robust.